sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ModelsAsyncTest.java

```diff
@@ -0,0 +1,137 @@
+        createModelsRunner(asyncClient, (modelsList) -> {
+            StepVerifier.create(asyncClient.createModels(modelsList))
+                .assertNext(createdModel -> {
+                    // When creating models, the service will not return the model itself, so it will be manually
+                    // assigned here and below to make comparisons easier
+                    createdModel.setModel(modelsList.get(0));
+                    createdModels.add(createdModel);
+                })
+                .assertNext(createdModel -> {
+                    createdModel.setModel(modelsList.get(1));
+                    createdModels.add(createdModel);
+                })
+                .assertNext(createdModel -> {
+                    createdModel.setModel(modelsList.get(2));
+                    createdModels.add(createdModel);
+                })
+                .verifyComplete();
+        });
+
+        for (int modelIndex = 0; modelIndex < createdModels.size(); modelIndex++) {
+            final ModelData expected = createdModels.get(modelIndex);
+
+            // Get the model
+            getModelRunner(expected.getId(), (modelId) -> {
+                StepVerifier.create(asyncClient.getModelWithResponse(modelId))
+                    .assertNext(retrievedModel -> {
+                        assertModelDataAreEqual(expected, retrievedModel.getValue());
+                    })
+                    .verifyComplete();
+            });
+
+            // Decommission the model
+            decommissionModelRunner(expected.getId(), (modelId) -> {
+                StepVerifier.create(asyncClient.decommissionModel(modelId))
+                    .verifyComplete();
+            });
+
+            // Get the model again to see if it was decommissioned as expected
+            getModelRunner(expected.getId(), (modelId) -> {
+                StepVerifier.create(asyncClient.getModel(modelId))
+                    .assertNext(retrievedModel -> {
+                        assertTrue(retrievedModel.isDecommissioned());
+                    })
+                    .verifyComplete();
+            });
+
+            // Delete the model
+            deleteModelRunner(expected.getId(), (modelId) -> {
+                StepVerifier.create(asyncClient.deleteModel(modelId))
+                    .verifyComplete();
+            });
+        }
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.digitaltwins.core.TestHelper#getTestParameters")
+    @Override
+    public void getModelThrowsIfModelDoesNotExist(HttpClient httpClient, DigitalTwinsServiceVersion serviceVersion) {
+        DigitalTwinsAsyncClient asyncClient = getAsyncClient(httpClient, serviceVersion);
+        final String nonExistantModelId = "urn:doesnotexist:fakemodel:1000";
+        StepVerifier.create(asyncClient.getModel(nonExistantModelId))
+            .verifyErrorSatisfies(ex -> assertRestException(ex, HttpURLConnection.HTTP_NOT_FOUND));
```

Comment:
I had been using `org.apache.http.HttpStatus` in the sample, but I like `java.net.HttpURLConnection` better; I'll make the switch.

Reply:
I see ```java.net.HttpURLConnection``` used in the app configuration SDK code, so I'd recommend it